src/libostree/ostree-repo.c

```diff
@@ -315,29 +362,38 @@ ostree_repo_get_remote_list_option (OstreeRepo   *self,
       return TRUE;
     }
 
-  remote = ost_repo_get_remote (self, remote_name, error);
-
+  remote = ost_repo_get_remote (self, remote_name, &temp_error);
   if (remote != NULL)
     {
-      g_auto(GStrv) value = NULL;
-      GError *local_error = NULL;
-
       value = g_key_file_get_string_list (remote->options,
                                           remote->group,
                                           option_name,
-                                          NULL, &local_error);
+                                          NULL, &temp_error);
 
       /* Default value if key not found is always NULL. */
-      if (g_error_matches (local_error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND))
-        g_clear_error (&local_error);
-
-      if (local_error == NULL)
+      if (g_error_matches (temp_error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND))
         {
-          ot_transfer_out_value (out_value, &value);
+          if (self->parent_repo != NULL)
+            return ostree_repo_get_remote_list_option (self->parent_repo,
+                                                       remote_name, option_name,
+                                                       out_value,
+                                                       error);
           ret = TRUE;
         }
+      else if (temp_error)
+        g_propagate_error (error, g_steal_pointer (&temp_error));
+      else
+        ret = TRUE;
     }
+  else if (self->parent_repo != NULL)
+    return ostree_repo_get_remote_list_option (self->parent_repo,
+                                               remote_name, option_name,
+                                               out_value,
+                                               error);
+  else
+    g_propagate_error (error, g_steal_pointer (&temp_error));
 
+  *out_value = g_steal_pointer (&value);
```

Comment:
Looks like we're doing the transfer even on error cases, which in this function is probably fine, but one of the reasons I structure the code as:

```
  *out_foo =  steal
  *out_bar = steal
   ret = TRUE;
out:
  return ret;
```

is to avoid stealing on errors.  Anyways it's okay as is probably but something to keep in mind elsewhere.